### ISSUE TITLE
Scala sandwich breaks zinc build

### ISSUE THREAD

**Issue by eatkins** [sbt/sbt#5797]
## steps

Publish latest sbt
run `sbt -Dsbt.version=1.4.0-SNAPSHOT scripted` in the zinc project

## problem
```
sbt:zinc Root> last zincScriptedJVM2_12/projectDependencies         
[error] java.lang.RuntimeException: scalaBinaryVersion mismatch: expected 2.12 but found 2.10                                           
[error]         at scala.sys.package$.error(package.scala:30)       
[error]         at sbt.Classpaths$.$anonfun$projectDependenciesTask$2(Defaults.scala:3490)                                              
[error]         at scala.collection.immutable.List.flatMap(List.scala:366)                                                              
[error]         at sbt.Classpaths$.$anonfun$projectDependenciesTask$1(Defaults.scala:3472)                                              
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)                                                               
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)                                           
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)                                                                   
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)                                                                     
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)                                                 
[error]         at sbt.Execute.work(Execute.scala:291)              
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)                                                                     
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:233)                          
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:59)                                                       
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)                                                             
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)                                              
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)                                                             
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)                                      
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)                                      
[error]         at java.lang.Thread.run(Thread.java:748)            
[error] (zincScriptedJVM2_12 / projectDependencies) scalaBinaryVersion mismatch: expected 2.12 but found 2.10    
```
## expectation

It compiles
## notes
zincScriptedJVM2_12 depends on all of the compiler bridges which all have different scala binary versions. Everything worked if I reverted f8139da192.

**Comment by eed3si9n** [comment]
This would be an extremely special situation (2.12 depending on 2.10), but I guess it is a regression I need to work around.

**Comment by eatkins** [comment]
Fixed in #5806
